### DISCUSSION TITLE
Code Annotations for JSON

### DISCUSSION THREAD

**Discussion by Rwarcards762** [squidfunk/mkdocs-material#3667] [Q&A] [answered]
Heyo! I know JSON does not formally support comments (other than "JSONC" or other variations.) Is there a way to utilize code annotations for JSON, or does the lack of formal comments in the language prevent them from being possible at all?

Thanks in advance! If it currently *isn't* possible, is it something that could be added (ie, support for `// this style` of comment would be supported for annotations?) or is that not feasible?

For a little context, I have a JSON structure that I'm trying to annotate as part of some documentation for setting up some software... requires a user/admin to fill in the JSON with their required information for the situation. Would be nice to have the annotations for each line rather than just using comments or something else.

**Comment by facelessuser** [comment]
Using JavaScript is usually sufficient for me to highlight JSON with comments. I'm not sure if there are edge cases that I'm not aware of, but that is usually what I use.

There may be a pygments 3rd party JSONC highlight syntax out there, but I've never bothered to search as the above has always worked for me.

**Reply by Rwarcards762** [reply]
I suppose that is one option, but using JS instead of JSON means giving up syntax highlighting between the key/value pairs... all just becomes one color (at least in the color theme we use.) 

This mostly turns into a question of "are annotations possible for non-recognized languages, OR for languages that do not support comments at all?"

**Reply by squidfunk** [reply]
It's currently not possible to add code annotations in languages without comments. I originally had plans to add this in the future, but it currently has very low priority. It is, however, theoretically possible. For JSON specifically, using the JavaScript lexer is a good alternative, albeit highlighting might not be as beautiful.

**Reply by Rwarcards762** [reply]
I kind of figured that was the case, but wanted to ask just in case. Thanks for the reply!

**Comment by facelessuser** [comment, marked as answer]
@Rwarcards762, it looks like comment support may be coming: https://github.com/pygments/pygments/pull/2049

So, the PR was merged on Jan 24th, and the last release was on Jan 6th. So, it is looking like the next release will have it.

EDIT: Next release of Pygments that is 😉

**Reply by Rwarcards762** [reply]
SWEET 😎 
Thanks for this update.
